agent/src/beerocks/slave/tasks/task_pool.h

```diff
@@ -0,0 +1,60 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2016-2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#ifndef _TASK_POOL_H_
+#define _TASK_POOL_H_
+
+#include "task.h"
+#include <unordered_map>
+
+namespace beerocks {
+
+/**
+ * @brief The TaskPool is single instance tasks container which responsible to run each
+ * task and allow passing messages and events from the Agent to the task.
+ * 
+ */
+class TaskPool {
+public:
+    /**
+     * @brief Add new task to the task pool.
+     * 
+     * @param new_task Shared pointer to the task.
+     */
+    void add_task(std::shared_ptr<Task> new_task);
```

Comment:
if you have an `add_task` method - you should have a `remove_task` method as well

Reply:
Currently, the task pool is designed for a single instance task which is implementing some Agent responsibility.
Meaning, that tasks should not be removed, at least of now POV.
If we will need such functionality, it will be added in the future.

Reply:
@arnout would appreciate your thoughts on this

Reply:
I like the "if we need it, it will be added" kind of approach.

It does make sense to think about it now - but adding a remove_task method is clearly something that can be done later.